CombinatorialGames/Game/Reductions.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2025 Aaron Liu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Liu
+-/
+import CombinatorialGames.Game.IGame
+import Mathlib.Algebra.Group.Pointwise.Set.Lattice
+
+/-!
+# Game reductions
+
+We prove that dominated moves can be deleted and reversible moves can be bypassed.
+-/
+
+universe u
+
+namespace IGame
+
+theorem equiv_of_dominated_left {u v r : Set IGame.{u}} [Small.{u} u] [Small.{u} v] [Small.{u} r]
+    (hu : ∀ g ∈ u, ∃ g' ∈ v, g ≤ g') : {u ∪ v | r}ᴵ ≈ {v | r}ᴵ := by
```

Comment:
I don't like how unions appear in the statement, since this makes it much harder to apply the lemma (you need to know the set has this precise form, or tediously rewrite it so that it does). 

Why not something like this, "{u | r} ≈ {v | r} when v is a subset of u, and every element in u is dominated by some element in v"? Though really, that would then be a special case of "{u | r} ≈ {v | r} when every element in u is dominated by some element in v all, and vice versa". And that would just be a special case of `equiv_of_exists_le` when the right sets of both games are equal. So maybe that's the statement you should prove instead?